tests/test_forms.py

```diff
@@ -1552,6 +1653,206 @@ async def validate_slot3(
+                    "intent": {"name": "inform", "confidence": 1.0},
+                    "entities": [{"entity": "name", "value": "Emily"}],
+                },
+            ),
+            SlotSet("slot2", "Emily"),
+        ],
+        False,
+        None,
+        {},
+        "action_listen",
+    )
+
+    domain = {
+        "slots": {
+            "slot1": {
+                "type": "any",
+                "mappings": [SlotMapping.from_entity(entity="name", intent="test")],
+            },
+            "slot2": {
+                "type": "any",
+                "mappings": [SlotMapping.from_entity(entity="name", intent="inform")],
+            },
+        }
+    }
+
+    dispatcher = CollectingDispatcher()
+    with pytest.warns(None) as warnings:
+        events = await validation_action.run(
+            dispatcher=dispatcher, tracker=tracker, domain=domain,
+        )
+
+    assert not warnings
+    assert events == [
+        SlotSet("slot2", "validated_value"),
+    ]
+
+
+async def test_validation_action_outside_forms_with_form_active_loop():
+    class TestSlotValidationAction(ValidationAction):
+        def global_slot_mappings(self) -> bool:
+            return True
+
+        def validate_slot1(
+            self,
+            slot_value: Any,
+            dispatcher: "CollectingDispatcher",
+            tracker: "Tracker",
+            domain: "DomainDict",
+        ) -> Dict[Text, Any]:
+            if slot_value == "Emily":
+                return {
+                    "slot1": "validated_value",
+                }
+            return {
+                "slot1": None,
+            }
+
+    validation_action = TestSlotValidationAction()
+
+    # tracker with active form
+    tracker = Tracker(
+        "default",
+        {},
+        {},
+        [
+            UserUttered(
+                "My name is Emily.",
+                parse_data={
+                    "intent": {"name": "inform", "confidence": 1.0},
+                    "entities": [{"entity": "name", "value": "Emily"}],
+                },
+            ),
+            SlotSet("slot1", "Emily"),
+        ],
+        False,
+        None,
+        {},
+        "action_listen",
+    )
+
+    domain = {
+        "slots": {
+            "slot1": {
+                "type": "any",
+                "mappings": [
+                    # this mapping means that the mapping should be validated by ValidationAction for `form1`
+                    {
+                        "type": "from_entity",
+                        "entity": "name",
+                        "conditions": [{"active_loop": "form1"}],
+                    }
+                ],
+            },
+        }
+    }
+
+    dispatcher = CollectingDispatcher()
+    with pytest.warns(None) as warnings:
+        events = await validation_action.run(
+            dispatcher=dispatcher, tracker=tracker, domain=domain,
+        )
+
+    assert not warnings
+    assert events[0] == SlotSet("slot1", "Emily")
```

Comment:
If `slot1` can only be set in an active loop and the tracker doesn't have an active loop running, shouldn't actually the output be an empty list?